### DISCUSSION TITLE
How to use integrals over an axis and explicit dependence on coordinates in a field?

### DISCUSSION THREAD

**Discussion by Divyoj-Singh** [zwicker-group/py-pde#75] [Q&A]
Hi!
First of all thanks for this great package!
I am trying to solve a pde in 3D which has terms like:

![image](https://user-images.githubusercontent.com/76565220/112485945-70674800-8da1-11eb-9161-76abf3549835.png)

and  `a(x+lcoz,y+sinz, z+ \pi,t)`

I tried the integrate function for the field but it calculates the integral for the whole lattice. 
I couldn't understand how to write the second term..
Can you please help me with this?

Thanks!!

**Comment by david-zwicker** [comment]
The partial integral over an axis is typically called a projection and is supported by the [`project`](https://py-pde.readthedocs.io/en/latest/packages/pde.fields.html?highlight=project#pde.fields.scalar.ScalarField.project) method. I hope this helps!

**Reply by Divyoj-Singh** [reply]
Yes, that worked, thanks!
